<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>View</class>
 <widget class="QMainWindow" name="View">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>815</width>
    <height>535</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>815</width>
    <height>530</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>815</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>View</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="settings" native="true">
    <property name="geometry">
     <rect>
      <x>535</x>
      <y>0</y>
      <width>280</width>
      <height>535</height>
     </rect>
    </property>
    <widget class="QLabel" name="label_06">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>380</y>
       <width>58</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Stepgen</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sBStepgen">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>380</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
    <widget class="Line" name="line_03">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>0</y>
       <width>251</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame
{
color: red;
}</string>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="lineWidth">
      <number>2</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="pBtNextGenerate">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>340</y>
       <width>251</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Next step</string>
     </property>
     <property name="autoRepeat">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_12">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>90</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Rows</string>
     </property>
    </widget>
    <widget class="Line" name="line_04">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>510</y>
       <width>251</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame
{
color: red;
}</string>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="lineWidth">
      <number>2</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="label_02">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>204</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Cols</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sBRowsMaze">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>90</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>50</number>
     </property>
     <property name="value">
      <number>10</number>
     </property>
    </widget>
    <widget class="QDoubleSpinBox" name="dSBChance">
     <property name="geometry">
      <rect>
       <x>82</x>
       <y>228</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true"/>
     </property>
     <property name="frame">
      <bool>true</bool>
     </property>
     <property name="value">
      <double>50.000000000000000</double>
     </property>
    </widget>
    <widget class="QPushButton" name="pBtCaveGenerate">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>260</y>
       <width>251</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Auto generation /OFF</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sBRowsCave">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>204</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>50</number>
     </property>
     <property name="value">
      <number>50</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_09">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>10</y>
       <width>71</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>18</pointsize>
       <bold>true</bold>
       <underline>true</underline>
      </font>
     </property>
     <property name="text">
      <string>Maze</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pBtOpenFile">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>440</y>
       <width>251</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>OpenFile</string>
     </property>
    </widget>
    <widget class="Line" name="line_02">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>120</y>
       <width>251</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame
{
color: red;
}</string>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="lineWidth">
      <number>2</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSpinBox" name="sBColsMaze">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>90</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>50</number>
     </property>
     <property name="value">
      <number>10</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_08">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>130</y>
       <width>71</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>18</pointsize>
       <bold>true</bold>
       <underline>true</underline>
      </font>
     </property>
     <property name="text">
      <string>Cave</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>90</y>
       <width>31</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Cols</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pBtCaveCreate">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>251</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Create</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>230</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>%</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_05">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>300</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Death</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_07">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>380</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Frequency</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sBDeath">
     <property name="geometry">
      <rect>
       <x>180</x>
       <y>300</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>7</number>
     </property>
     <property name="value">
      <number>4</number>
     </property>
    </widget>
    <widget class="QPushButton" name="pBtMazeGenerate">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>251</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Create ideal maze</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_01">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>204</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Rows</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_03">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>224</y>
       <width>58</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>Chance</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pBtSave">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>480</y>
       <width>251</width>
       <height>32</height>
      </rect>
     </property>
     <property name="text">
      <string>Save</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sBBirth">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>300</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>7</number>
     </property>
     <property name="value">
      <number>4</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_04">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>300</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Birth</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="sBColsCave">
     <property name="geometry">
      <rect>
       <x>170</x>
       <y>204</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>50</number>
     </property>
     <property name="value">
      <number>50</number>
     </property>
    </widget>
    <widget class="Line" name="line_01">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>10</y>
       <width>21</width>
       <height>501</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <bold>true</bold>
      </font>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame[frameShape=&quot;5&quot;] /* QFrame::Line == 0x0005 */
{
    color: green;
}</string>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="lineWidth">
      <number>2</number>
     </property>
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
    <widget class="QSpinBox" name="sBFrequency">
     <property name="geometry">
      <rect>
       <x>220</x>
       <y>380</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="value">
      <number>1</number>
     </property>
    </widget>
    <widget class="Line" name="line_3">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>420</y>
       <width>251</width>
       <height>16</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame
{
color: red;
}</string>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="lineWidth">
      <number>2</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QWidget" name="field" native="true">
    <property name="geometry">
     <rect>
      <x>15</x>
      <y>15</y>
      <width>504</width>
      <height>504</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>500</width>
      <height>500</height>
     </size>
    </property>
    <property name="maximumSize">
     <size>
      <width>520</width>
      <height>520</height>
     </size>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <widget class="QWidget" name="verticalLayoutWidget">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>511</width>
       <height>511</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="layout"/>
    </widget>
   </widget>
  </widget>
 </widget>
 <tabstops>
  <tabstop>sBRowsMaze</tabstop>
  <tabstop>sBColsMaze</tabstop>
  <tabstop>pBtMazeGenerate</tabstop>
  <tabstop>sBRowsCave</tabstop>
  <tabstop>sBColsCave</tabstop>
  <tabstop>dSBChance</tabstop>
  <tabstop>sBBirth</tabstop>
  <tabstop>sBDeath</tabstop>
  <tabstop>sBStepgen</tabstop>
  <tabstop>sBFrequency</tabstop>
  <tabstop>pBtCaveCreate</tabstop>
  <tabstop>pBtCaveGenerate</tabstop>
  <tabstop>pBtNextGenerate</tabstop>
  <tabstop>pBtOpenFile</tabstop>
  <tabstop>pBtSave</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
